src/test/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMBuilderTest.java

```diff
@@ -1230,8 +1230,8 @@ public void given__cloud_pullHead_rev_userkey__when__build__then__scmBuilt() thr
         PullRequestSCMRevision revision = new PullRequestSCMRevision(
                 head, "deadbeefcafebabedeadbeefcafebabedeadbeef", "cafebabedeadbeefcafebabedeadbeefcafebabe");
         createGitHubSCMSourceForTest(false, null);
-        source.setCredentialsId("user-key");
         GitHubSCMBuilder instance = new GitHubSCMBuilder(source, head, revision);
+        instance.withCredentials("user-key", GitHubSCMBuilder.SSH);
```

Comment:
Nit: prefer non deprecated API usage
```suggestion
        GitHubSCMBuilder instance = new GitHubSCMBuilder(source, head, revision)
                .withCredentials("user-key")
                .withResolver(GitHubSCMBuilder.SSH);
```